internal-messaging/message/cc/DataMessage.cc

```diff
@@ -110,75 +59,15 @@ DataMessage::DataMessage(char* flat)
 
     // Find key value pairs
     i++;
-    int key;
-    while(flat[i] != '\3'){
-        //Get key
-        while(flat[i] != '~'){
-            sprintf(integer_string, "%c", flat[i]);
-            strcat(hex_string, integer_string); 
-            i++;
-        }
-        i++;
-        try {
-            key = std::stoi(hex_string,nullptr,16);
-        } catch (std::exception const &e) {}
-        memset(long_string, 0, 64);
-        memset(hex_string, 0, 32);
-        // Get value
-        while(flat[i] != '|'){
-            sprintf(integer_string, "%c", flat[i]);
-            strcat(hex_string, integer_string); 
-            i++;
-        }
-        try {
-            if(!number_value(hex_string)){
-                std::string value = hex_string;
-                contents_.AddKeyValuePair(key, value);
-            }
-            else if(strchr(hex_string, 46)){ // If hex_string contains a '.'
-                float value = std::stof(hex_string);
-                contents_.AddKeyValuePair(key, value);
-            }
-            else{
-                int value = std::stoul(hex_string,nullptr,16);
-                contents_.AddKeyValuePair(key, value);
-            }
-        } catch (std::exception const &e) {}
-        memset(long_string, 0, 64);
-        memset(hex_string, 0, 32);
-        i++;
-    }
+    i = BuildContents(flat, i);
```

Comment:
Hmm... It seems here that the behavior of `flatten` slightly is different between `DataMessage` and `CommandMessage` where `BuildContents` is called in the constructor of `DataMessage` compared to in `CommandMessage` where it is called in `flatten()`. Is this desired? It might be ideal to move `BuildContents` to either both be in the constructor or both in the `flatten()`

Reply:
BuildContents should be used in the constructors, not Flatten. However I don't see a BuildContents in the Flatten method of CommandMessage. am I missing something? 

BuildContents is called on line 38 of CommandMessage.cc, which is within its constructor

Reply:
Ah my bad, I must have read this too quickly. You are correct here sorry